### ISSUE TITLE
Getting capture path from extension does not work

### ISSUE THREAD

**Issue by kckeiks** [tokio-rs/axum#2029]
- [ ] I have looked for existing issues (including closed) about this

### Version

```
axum v0.6.18
axum-core v0.3.4
```
### Platform

MacOS 13.0

**Comment by kckeiks** [comment]
It seems to work with`extract_parts_with_state`.

**Comment by jplatte** [comment]
Are you looking for the whole request path? That would be `req.uri().path()`. `Path` is for extracting path parameters, and it's not possible for axum to put it in the request extensions since it has a generic parameter chosen by the user independently of the routing.

**Comment by kckeiks** [comment]
> Are you looking for the whole request path? That would be `req.uri().path()`. `Path` is for extracting path parameters, and it's not possible for axum to put it in the request extensions since it has a generic parameter chosen by the user independently of the routing.

Not the whole request path but the captured value in the path. In the example above, I'd like the value of `:id`.

**Comment by jplatte** [comment]
Right, there is no way to do that without async at the moment. I think it would be easy enough to make it possible, but not sure if it's really worth it.

**Comment by kckeiks** [comment]
That's just an example. I'm my application I'm using a Future that will be polled.
